mux.go

```diff
@@ -40,6 +40,23 @@ func Group(s *Server, path string) *Server {
 	}
 	newServer.mainRouter = s
 
+	baseRoute := BaseRoute{
+		Params:    make(map[string]OpenAPIParam),
+		Operation: openapi3.NewOperation(),
+	}
+	for _, option := range options {
+		option(&baseRoute)
+	}
```

Comment:
nitpick: could save a few lines by creating a helper function for `WithRouteOptions` and this. Would be nice if we ever change anything about this.